java/org/contikios/cooja/contikimote/interfaces/ContikiBeeper.java

```diff
@@ -86,9 +81,12 @@ public boolean isBeeping() {
   @Override
   public void doActionsAfterTick() {
     if (moteMem.getByteValueOf("simBeeped") == 1) {
-      this.setChanged();
-      this.notifyObservers(mote);
-
+      var now = mote.getSimulation().getSimulationTime();
+      for (var label : labels.values()) {
+        label.setText("Last beep at time: " + now);
+        // Beep on speakers.
+        Toolkit.getDefaultToolkit().beep();
+      }
```

Comment:
It looks good, but I suggest switching to the event dispatch thread when updating the labels. Using the HashMap only from the dispatch thread will also eliminate any risk for concurrent modifications.